pkg/clusters/addons/kongargo/addon.go

```diff
@@ -0,0 +1,227 @@
+package kongargo
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/argocd"
+)
+
+const (
+	// AddonName indicates the unique name of this addon.
+	AddonName clusters.AddonName = "kong-argo"
+
+	// DefaultNamespace indicates the default namespace this addon will be deployed to.
+	DefaultNamespace = "kong-argo"
+
+	defaultProject = "ktf-kong"
+
+	defaultVersion = "2.25.0"
+
+	defaultRelease = "ktf-argo"
+
+	defaultAppName = "kong"
+)
+
+type Addon struct {
+	name      string
+	namespace string
+	version   string
+	project   string
+	release   string
+	appName   string
+}
+
+func New() clusters.Addon {
+	return &Addon{}
+}
+
+func (a *Addon) Name() clusters.AddonName {
+	return AddonName
+}
+
+func (a *Addon) Dependencies(_ context.Context, _ clusters.Cluster) []clusters.AddonName {
+	return []clusters.AddonName{
+		argocd.AddonName,
+	}
+}
+
+func getArgo(cluster clusters.Cluster) (*argocd.Addon, error) {
+	addon, err := cluster.GetAddon(argocd.AddonName)
+	if err != nil {
+		return nil, err
+	}
+	argo, ok := addon.(*argocd.Addon)
+	if !ok {
+		return nil, fmt.Errorf("argocd addon is not actually an argocd addon")
+	}
+	return argo, nil
+}
+
+func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
+	// wait for dependency addons to be ready first
+	if err := clusters.WaitForAddonDependencies(ctx, cluster, a); err != nil {
+		return fmt.Errorf("failure waiting for addon dependencies: %w", err)
+	}
+
+	ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: a.namespace}}
+	if _, err := cluster.Client().CoreV1().Namespaces().Create(ctx, &ns, metav1.CreateOptions{}); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+
+	argo, err := getArgo(cluster)
+	if err != nil {
+		return err
```

Comment:
I think we could make use of error wrapping here to provide better context in case of failures (it applies to all places in this PR where we return err without wrapping it).
```suggestion
		return fmt.Errorf("failed to get ArgoCD addon: %w", err)
```